--- xorg-2.eclass	2017-03-07 07:50:11.327378947 +0200
+++ xorg-3.eclass	2017-03-16 18:28:04.028393761 +0200
@@ -1,7 +1,7 @@
-# Copyright 1999-2015 Gentoo Foundation
+# Copyright 1999-2017 Gentoo Foundation
 # Distributed under the terms of the GNU General Public License v2
 
-# @ECLASS: xorg-2.eclass
+# @ECLASS: xorg-3.eclass
 # @MAINTAINER:
 # x11@gentoo.org
 # @AUTHOR:
@@ -25,17 +25,6 @@
 	XORG_EAUTORECONF="yes"
 fi
 
-# If we're a font package, but not the font.alias one
-FONT_ECLASS=""
-if [[ ${PN} == font* \
-	&& ${CATEGORY} = media-fonts \
-	&& ${PN} != font-alias \
-	&& ${PN} != font-util ]]; then
-	# Activate font code in the rest of the eclass
-	FONT="yes"
-	FONT_ECLASS="font"
-fi
-
 # @ECLASS-VARIABLE: XORG_MULTILIB
 # @DESCRIPTION:
 # If set to 'yes', the multilib support for package will be enabled. Set
@@ -43,21 +32,20 @@
 : ${XORG_MULTILIB:="no"}
 
 # we need to inherit autotools first to get the deps
-inherit autotools autotools-utils eutils libtool multilib toolchain-funcs \
-	flag-o-matic ${FONT_ECLASS} ${GIT_ECLASS}
+inherit autotools libtool multilib toolchain-funcs \
+	flag-o-matic ${GIT_ECLASS}
 
-if [[ ${XORG_MULTILIB} == yes ]]; then
-	inherit autotools-multilib
-fi
+#if [[ ${XORG_MULTILIB} == yes ]]; then
+#	inherit autotools-multilib
+#fi
 
-EXPORTED_FUNCTIONS="src_unpack src_compile src_install pkg_postinst pkg_postrm"
 case "${EAPI:-0}" in
-	3|4|5) EXPORTED_FUNCTIONS="${EXPORTED_FUNCTIONS} src_prepare src_configure" ;;
+	6) ;;
 	*) die "EAPI=${EAPI} is not supported" ;;
 esac
 
 # exports must be ALWAYS after inherit
-EXPORT_FUNCTIONS ${EXPORTED_FUNCTIONS}
+EXPORT_FUNCTIONS src_prepare src_configure src_install pkg_postinst pkg_postrm
 
 IUSE=""
 HOMEPAGE="https://www.x.org/wiki/"
@@ -100,7 +88,7 @@
 : ${XORG_PACKAGE_NAME:=${PN}}
 
 if [[ -n ${GIT_ECLASS} ]]; then
-	: ${EGIT_REPO_URI:="git://anongit.freedesktop.org/xorg/${XORG_MODULE}${XORG_PACKAGE_NAME} http://anongit.freedesktop.org/git/xorg/${XORG_MODULE}${XORG_PACKAGE_NAME}"}
+	: ${EGIT_REPO_URI:="git://anongit.freedesktop.org/xorg/${XORG_MODULE}${XORG_PACKAGE_NAME} https://anongit.freedesktop.org/git/xorg/${XORG_MODULE}${XORG_PACKAGE_NAME}"}
 elif [[ -n ${XORG_BASE_INDIVIDUAL_URI} ]]; then
 	SRC_URI="${XORG_BASE_INDIVIDUAL_URI}/${XORG_MODULE}${P}.tar.bz2"
 fi
@@ -133,33 +121,6 @@
 unset EAUTORECONF_DEPENDS
 unset EAUTORECONF_DEPEND
 
-if [[ ${FONT} == yes ]]; then
-	RDEPEND+=" media-fonts/encodings
-		x11-apps/mkfontscale
-		x11-apps/mkfontdir"
-	PDEPEND+=" media-fonts/font-alias"
-	DEPEND+=" >=media-fonts/font-util-1.2.0"
-
-	# @ECLASS-VARIABLE: FONT_DIR
-	# @DESCRIPTION:
-	# If you're creating a font package and the suffix of PN is not equal to
-	# the subdirectory of /usr/share/fonts/ it should install into, set
-	# FONT_DIR to that directory or directories. Set before inheriting this
-	# eclass.
-	[[ -z ${FONT_DIR} ]] && FONT_DIR=${PN##*-}
-
-	# Fix case of font directories
-	FONT_DIR=${FONT_DIR/ttf/TTF}
-	FONT_DIR=${FONT_DIR/otf/OTF}
-	FONT_DIR=${FONT_DIR/type1/Type1}
-	FONT_DIR=${FONT_DIR/speedo/Speedo}
-
-	# Set up configure options, wrapped so ebuilds can override if need be
-	[[ -z ${FONT_OPTIONS} ]] && FONT_OPTIONS="--with-fontdir=\"${EPREFIX}/usr/share/fonts/${FONT_DIR}\""
-
-	[[ ${PN##*-} = misc || ${PN##*-} = 75dpi || ${PN##*-} = 100dpi || ${PN##*-} = cyrillic ]] && IUSE+=" nls"
-fi
-
 # If we're a driver package, then enable DRIVER case
 [[ ${PN} == xf86-video-* || ${PN} == xf86-input-* ]] && DRIVER="yes"
 
@@ -292,13 +253,7 @@
 fi
 
 if [[ ${XORG_MODULE_REBUILD} == yes ]]; then
-	case ${EAPI} in
-		3|4)
-			;;
-		*)
-			RDEPEND+=" x11-base/xorg-server:="
-			;;
-	esac
+	RDEPEND+=" x11-base/xorg-server:="
 fi
 
 DEPEND+=" ${COMMON_DEPEND}"
@@ -313,112 +268,28 @@
 debug-print "${LINENO} ${ECLASS} ${FUNCNAME}: RDEPEND=${RDEPEND}"
 debug-print "${LINENO} ${ECLASS} ${FUNCNAME}: PDEPEND=${PDEPEND}"
 
-# @FUNCTION: xorg-2_pkg_setup
-# @DESCRIPTION:
-# Setup prefix compat
-xorg-2_pkg_setup() {
-	debug-print-function ${FUNCNAME} "$@"
-
-	[[ ${FONT} == yes ]] && font_pkg_setup "$@"
-}
-
-# @FUNCTION: xorg-2_src_unpack
-# @DESCRIPTION:
-# Simply unpack source code.
-xorg-2_src_unpack() {
-	debug-print-function ${FUNCNAME} "$@"
-
-	if [[ -n ${GIT_ECLASS} ]]; then
-		git-r3_src_unpack
-	else
-		unpack ${A}
-	fi
-
-	[[ -n ${FONT_OPTIONS} ]] && einfo "Detected font directory: ${FONT_DIR}"
-}
-
-# @FUNCTION: xorg-2_patch_source
+# @FUNCTION: xorg-3_src_prepare
 # @DESCRIPTION:
-# Apply all patches
-xorg-2_patch_source() {
+# Prepare a package after unpacking, performing all X-related tasks.
+xorg-3_src_prepare() {
 	debug-print-function ${FUNCNAME} "$@"
 
-	# Use standardized names and locations with bulk patching
-	# Patch directory is ${WORKDIR}/patch
-	# See epatch() in eutils.eclass for more documentation
-	EPATCH_SUFFIX=${EPATCH_SUFFIX:=patch}
-
-	[[ -d "${EPATCH_SOURCE}" ]] && epatch
-}
-
-# @FUNCTION: xorg-2_reconf_source
-# @DESCRIPTION:
-# Run eautoreconf if necessary, and run elibtoolize.
-xorg-2_reconf_source() {
-	debug-print-function ${FUNCNAME} "$@"
+	default
 
 	case ${CHOST} in
 		*-aix* | *-winnt*)
-			# some hosts need full eautoreconf
-			[[ -e "./configure.ac" || -e "./configure.in" ]] \
-				&& AUTOTOOLS_AUTORECONF=1
+			[[ -e "./configure.ac" ]] && eautoreconf
 			;;
 		*)
-			# elibtoolize required for BSD
-			[[ ${XORG_EAUTORECONF} != no && ( -e "./configure.ac" || -e "./configure.in" ) ]] \
-				&& AUTOTOOLS_AUTORECONF=1
+			[[ ${XORG_EAUTORECONF} == 'yes' ]] && eautoreconf
 			;;
 	esac
 }
 
-# @FUNCTION: xorg-2_src_prepare
-# @DESCRIPTION:
-# Prepare a package after unpacking, performing all X-related tasks.
-xorg-2_src_prepare() {
-	debug-print-function ${FUNCNAME} "$@"
-
-	xorg-2_patch_source
-	xorg-2_reconf_source
-	autotools-utils_src_prepare "$@"
-}
-
-# @FUNCTION: xorg-2_font_configure
-# @DESCRIPTION:
-# If a font package, perform any necessary configuration steps
-xorg-2_font_configure() {
-	debug-print-function ${FUNCNAME} "$@"
-
-	if has nls ${IUSE//+} && ! use nls; then
-		if grep -q -s "disable-all-encodings" ${ECONF_SOURCE:-.}/configure; then
-			FONT_OPTIONS+="
-				--disable-all-encodings"
-		else
-			FONT_OPTIONS+="
-				--disable-iso8859-2
-				--disable-iso8859-3
-				--disable-iso8859-4
-				--disable-iso8859-5
-				--disable-iso8859-6
-				--disable-iso8859-7
-				--disable-iso8859-8
-				--disable-iso8859-9
-				--disable-iso8859-10
-				--disable-iso8859-11
-				--disable-iso8859-12
-				--disable-iso8859-13
-				--disable-iso8859-14
-				--disable-iso8859-15
-				--disable-iso8859-16
-				--disable-jisx0201
-				--disable-koi8-r"
-		fi
-	fi
-}
-
-# @FUNCTION: xorg-2_flags_setup
+# @FUNCTION: xorg-3_flags_setup
 # @DESCRIPTION:
 # Set up CFLAGS for a debug build
-xorg-2_flags_setup() {
+xorg-3_flags_setup() {
 	debug-print-function ${FUNCNAME} "$@"
 
 	# Win32 require special define
@@ -434,38 +305,19 @@
 	fi
 }
 
-# @FUNCTION: xorg-2_src_configure
+# @FUNCTION: xorg-3_src_configure
 # @DESCRIPTION:
 # Perform any necessary pre-configuration steps, then run configure
-xorg-2_src_configure() {
+xorg-3_src_configure() {
 	debug-print-function ${FUNCNAME} "$@"
 
-	xorg-2_flags_setup
+	xorg-3_flags_setup
 
 	# @VARIABLE: XORG_CONFIGURE_OPTIONS
 	# @DESCRIPTION:
 	# Array of an additional options to pass to configure.
 	# @DEFAULT_UNSET
-	if [[ $(declare -p XORG_CONFIGURE_OPTIONS 2>&-) != "declare -a"* ]]; then
-		# fallback to CONFIGURE_OPTIONS, deprecated.
-		if [[ -n "${CONFIGURE_OPTIONS}" ]]; then
-			eqawarn "CONFIGURE_OPTIONS are deprecated. Please migrate to XORG_CONFIGURE_OPTIONS"
-			eqawarn "to preserve namespace."
-		fi
-
-		local xorgconfadd=(${CONFIGURE_OPTIONS} ${XORG_CONFIGURE_OPTIONS})
-	else
-		local xorgconfadd=("${XORG_CONFIGURE_OPTIONS[@]}")
-	fi
-
-	[[ -n "${FONT}" ]] && xorg-2_font_configure
-
-	# Check if package supports disabling of dep tracking
-	# Fixes warnings like:
-	#    WARNING: unrecognized options: --disable-dependency-tracking
-	if grep -q -s "disable-depencency-tracking" ${ECONF_SOURCE:-.}/configure; then
-		local dep_track="--disable-dependency-tracking"
-	fi
+	local xorgconfadd=("${XORG_CONFIGURE_OPTIONS[@]}")
 
 	# Check if package supports disabling of selective -Werror=...
 	if grep -q -s "disable-selective-werror" ${ECONF_SOURCE:-.}/configure; then
@@ -473,138 +325,41 @@
 	fi
 
 	local myeconfargs=(
-		${dep_track}
 		${selective_werror}
-		${FONT_OPTIONS}
 		"${xorgconfadd[@]}"
 	)
 
-	if [[ ${XORG_MULTILIB} == yes ]]; then
-		autotools-multilib_src_configure "$@"
-	else
-		autotools-utils_src_configure "$@"
-	fi
+	econf "${myeconfargs[@]}"
+#	if [[ ${XORG_MULTILIB} == yes ]]; then
+#		autotools-multilib_src_configure "$@"
+#	else
+#		autotools-utils_src_configure "$@"
+#	fi
 }
 
-# @FUNCTION: xorg-2_src_compile
-# @DESCRIPTION:
-# Compile a package, performing all X-related tasks.
-xorg-2_src_compile() {
-	debug-print-function ${FUNCNAME} "$@"
-
-	if [[ ${XORG_MULTILIB} == yes ]]; then
-		autotools-multilib_src_compile "$@"
-	else
-		autotools-utils_src_compile "$@"
-	fi
-}
 
-# @FUNCTION: xorg-2_src_install
+# @FUNCTION: xorg-3_src_install
 # @DESCRIPTION:
 # Install a built package to ${D}, performing any necessary steps.
 # Creates a ChangeLog from git if using live ebuilds.
-xorg-2_src_install() {
+xorg-3_src_install() {
 	debug-print-function ${FUNCNAME} "$@"
 
-	local install_args=( docdir="${EPREFIX}/usr/share/doc/${PF}" )
-
-	if [[ ${CATEGORY} == x11-proto ]]; then
-		install_args+=(
-			${PN/proto/}docdir="${EPREFIX}/usr/share/doc/${PF}"
-		)
-	fi
+#	if [[ ${XORG_MULTILIB} == yes ]]; then
+#		autotools-multilib_src_install "${install_args[@]}"
+#	else
+#		autotools-utils_src_install "${install_args[@]}"
+#	fi
+
+#	if [[ -n ${GIT_ECLASS} ]]; then
+#		pushd "${EGIT_STORE_DIR}/${EGIT_CLONE_DIR}" > /dev/null || die
+#		git log ${EGIT_COMMIT} > "${S}"/ChangeLog
+#		popd > /dev/null || die
+#	fi
 
-	if [[ ${XORG_MULTILIB} == yes ]]; then
-		autotools-multilib_src_install "${install_args[@]}"
-	else
-		autotools-utils_src_install "${install_args[@]}"
-	fi
-
-	if [[ -n ${GIT_ECLASS} ]]; then
-		pushd "${EGIT_STORE_DIR}/${EGIT_CLONE_DIR}" > /dev/null || die
-		git log ${EGIT_COMMIT} > "${S}"/ChangeLog
-		popd > /dev/null || die
-	fi
-
-	if [[ -e "${S}"/ChangeLog ]]; then
-		dodoc "${S}"/ChangeLog || die "dodoc failed"
-	fi
+	default
 
 	# Don't install libtool archives (even for modules)
-	prune_libtool_files --all
-
-	[[ -n ${FONT} ]] && remove_font_metadata
-}
-
-# @FUNCTION: xorg-2_pkg_postinst
-# @DESCRIPTION:
-# Run X-specific post-installation tasks on the live filesystem. The
-# only task right now is some setup for font packages.
-xorg-2_pkg_postinst() {
-	debug-print-function ${FUNCNAME} "$@"
-
-	if [[ -n ${FONT} ]]; then
-		create_fonts_scale
-		create_fonts_dir
-		font_pkg_postinst "$@"
-	fi
-}
-
-# @FUNCTION: xorg-2_pkg_postrm
-# @DESCRIPTION:
-# Run X-specific post-removal tasks on the live filesystem. The only
-# task right now is some cleanup for font packages.
-xorg-2_pkg_postrm() {
-	debug-print-function ${FUNCNAME} "$@"
-
-	if [[ -n ${FONT} ]]; then
-		# if we're doing an upgrade, postinst will do
-		if [[ ${EAPI} -lt 4 || -z ${REPLACED_BY_VERSION} ]]; then
-			create_fonts_scale
-			create_fonts_dir
-			font_pkg_postrm "$@"
-		fi
-	fi
-}
-
-# @FUNCTION: remove_font_metadata
-# @DESCRIPTION:
-# Don't let the package install generated font files that may overlap
-# with other packages. Instead, they're generated in pkg_postinst().
-remove_font_metadata() {
-	debug-print-function ${FUNCNAME} "$@"
-
-	if [[ ${FONT_DIR} != Speedo && ${FONT_DIR} != CID ]]; then
-		einfo "Removing font metadata"
-		rm -rf "${ED}"/usr/share/fonts/${FONT_DIR}/fonts.{scale,dir,cache-1}
-	fi
-}
-
-# @FUNCTION: create_fonts_scale
-# @DESCRIPTION:
-# Create fonts.scale file, used by the old server-side fonts subsystem.
-create_fonts_scale() {
-	debug-print-function ${FUNCNAME} "$@"
-
-	if [[ ${FONT_DIR} != Speedo && ${FONT_DIR} != CID ]]; then
-		ebegin "Generating fonts.scale"
-			mkfontscale \
-				-a "${EROOT}/usr/share/fonts/encodings/encodings.dir" \
-				-- "${EROOT}/usr/share/fonts/${FONT_DIR}"
-		eend $?
-	fi
-}
-
-# @FUNCTION: create_fonts_dir
-# @DESCRIPTION:
-# Create fonts.dir file, used by the old server-side fonts subsystem.
-create_fonts_dir() {
-	debug-print-function ${FUNCNAME} "$@"
-
-	ebegin "Generating fonts.dir"
-			mkfontdir \
-				-e "${EROOT}"/usr/share/fonts/encodings \
-				-e "${EROOT}"/usr/share/fonts/encodings/large \
-				-- "${EROOT}/usr/share/fonts/${FONT_DIR}"
-	eend $?
+	#prune_libtool_files --all
+	find "${D}" -name '*.la' -delete || die
 }
